src/prefect/agent.py

```diff
@@ -91,6 +91,23 @@ def __init__(
             self.default_infrastructure = Process()
             self.default_infrastructure_document_id = None
 
+    async def _select_default_agent_queues(
+        self, client: PrefectClient, queues: List[WorkQueue]
+    ) -> List[WorkQueue]:
+        """
+        Select only those WorkQueues that match the default work pool type, "prefect-agent"
+        """
+        # get the pools for each matched queue
+        wp_filter = WorkPoolFilter(
```

Comment:
If I remember correctly, if no work pool is provided to the agent, it should only poll the `default-agent-pool` work pool. I think the filter used in the API call should be updated to reflect that.

Reply:
I butted heads with something like this while I was writing my tests.

It looks like the server enforces that the query only polls `default-agent-pool` iff that pool exists. In the case it doesn't exist the query hits the `WorkQueue` table directly, and is the case where I think we'd want to do the filtering on all the `WorkPool` names. [Relevant code here](https://github.com/PrefectHQ/prefect/blob/271b76dadc293dad595e6fa326574b86d5649ae2/src/prefect/server/models/work_queues.py#L143-L152).

I'm not sure if all agents will have the `default-agent-pool` created/ if we have that condition for backwards compatibility. Do you think it's alright to change the client behavior to always return results from the default work pool?

Reply:
Hitting the `work_queue` table directly is legacy behavior. We kept it in to facilitate the migration to work pools, and it could be removed/updated now. Every server/workspace gets a `default-agent-pool` work pool, which is why it acts as the default.

Reply:
I just pushed some changes that removes the legacy behavior from the client's side. Let me know what you think!